eng/doc/fips/UserGuide.md

```diff
@@ -645,6 +658,204 @@ sha512.Sum512_224 is not FIPS compliant.
 
 sha512.Sum512_256 is not FIPS compliant.
 
+### [crypto/rsa](https://pkg.go.dev/crypto/rsa)
+
+Package rsa implements RSA encryption as specified in PKCS #1 and RFC 8017.
+
+#### func [DecryptOAEP](https://pkg.go.dev/crypto/rsa#DecryptOAEP)
+
+```go
+func rsa.DecryptOAEP(h hash.Hash, rand io.Reader, priv *rsa.PrivateKey, ciphertext []byte, label []byte) ([]byte, error)
+```
+
+DecryptOAEP decrypts ciphertext using RSA-OAEP.
+
+**Parameters**
+
+`h` must be the result of one of the following functions in order to be FIPS compliant: sha1.New(), sha224.New(), sha256.New(), sha384.New(), or sha512.New().
+If this invariant is not met, DecryptOAEP won't be FIPS compliant but still will decrypt the message.
+
+`rand` is not used.
+
+**Return values**
+
+The decrypted buffer generated using [EVP_PKEY_decrypt] with `RSA_PKCS1_OAEP_PADDING`.
+
+#### func [DecryptPKCS1v15](https://pkg.go.dev/crypto/rsa#DecryptPKCS1v15)
+
+```go
+func rsa.DecryptPKCS1v15(rand io.Reader, priv *rsa.PrivateKey, ciphertext []byte) ([]byte, error)
+```
+
+DecryptPKCS1v15 decrypts a plaintext using RSA and the padding scheme from PKCS #1 v1.5.
+
+**Parameters**
+
+`rand` is not used.
+
+**Return values**
+
+The plaintext message generated using [EVP_PKEY_decrypt] with `RSA_PKCS1_PADDING`.
+
+#### func [DecryptPKCS1v15SessionKey](https://pkg.go.dev/crypto/rsa#DecryptPKCS1v15SessionKey)
+
+```go
+func rsa.DecryptPKCS1v15SessionKey(rand io.Reader, priv *PrivateKey, ciphertext []byte, key []byte) error
+```
+
+DecryptPKCS1v15SessionKey decrypts a session key using RSA and the padding scheme from PKCS #1 v1.5.
+
+**Parameters**
+
+`rand` is not used.
+
+The plaintext message generated using [EVP_PKEY_decrypt] with `RSA_PKCS1_PADDING` is copied into `key`.
+
+#### func [EncryptPKCS1v15](https://pkg.go.dev/crypto/rsa#EncryptPKCS1v15)
+
+```go
+func rsa.SignPKCS1v15(rand io.Reader, priv *rsa.PrivateKey, hash crypto.Hash, hashed []byte) ([]byte, error)
+```
+
+SignPKCS1v15 calculates the signature of hashed using RSASSA-PKCS1-V1_5-SIGN from RSA PKCS #1 v1.5.
+
+**Parameters**
+
+`rand` is not used.
+
+`hash` can be one of the following values: crypto.MD5, crypto.MD5SHA1, crypto.SHA1, crypto.SHA224, crypto.SHA256, rypto.SHA384, or crypto.SHA512. Else SignPKCS1v15 will fail.

```

Comment:
I would say "must be", to make it clear from the start that they're the only options.

I also think it might be a good idea to use the same wording for "Else" as "If this invariant is not met" elsewhere, to make it easier to spot these clauses when reading through. (I don't have a strong opinion on whether it'd be better to change "If this invariant is not met" to "Else" or the other way around. I'm not used to seeing "Else" at the beginning of a sentence, but it certainly makes sense.)

```suggestion
`hash` must be one of the following values: crypto.MD5, crypto.MD5SHA1, crypto.SHA1, crypto.SHA224, crypto.SHA256, rypto.SHA384, or crypto.SHA512. Else SignPKCS1v15 will fail.
```